webapp/components/experiments/EditExperiment.js

```diff
@@ -49,21 +227,242 @@ const EditExperiment = () => {
       });
   };
 
+  const [selectionParams, setSelectionParams] = useState([])
   const viewParameters = () => {
     getParameters().then((parameters) => {
       // netParams JSON dict
-      console.log(parameters);
+      setSelectionParams(Object.keys(parameters));
     });
   };
 
+  useEffect(() => {
+    viewParameters();
+  }, []);
+
+  const handleChange = (event, parameter, index) => {
+    const newParam = parameter.inGroup ?  [...groupParameters] : [...parameters];
+    const newValues = event.target.value === RANGE ? { min: 0, max: 0, step: 0, type: RANGE } : { values: [], type: LIST, val: '' }
+    newParam[index] = { ...parameter, ...newValues };
+    parameter.inGroup ? setGroupParameters(newParam) : setParameters(newParam);
+  };
+
+  const handleParamSelection = (val, parameter, index) => {
+    const newParam = parameter.inGroup ?  [...groupParameters] : [...parameters];
+    newParam[index] = { ...parameter, 'mapsTo': val };
+    parameter.inGroup ? setGroupParameters(newParam) : setParameters(newParam);
+  }
+
+  const ParameterMenu = (props) => {
```

Comment:
Would avoid to define nested components, makes it less readable & hard to reuse, the component is tightly coupled & already quite large. 

One example for tight coupling: you define `handleClose` in `ParameterMenu` but use it e.g. in `addToGroup` method that is define outside of this scope, my IDE marks this, it only works atm due to the order of the function declarations. And the other way around, the inner component calls method of the outer component

Avoid large render methods that have more than ~100 lines of code & split this up into functions to increase readability & abstraction.

You can move components that have their own state to a separate file and smaller stateless components to the global level in the same file